### ISSUE TITLE
Creating Grid Views via Project Creator API

### ISSUE THREAD

**Issue by tischi** [mobie/mobie-viewer-fiji#667]
@CamachoDejay

It loos like you like to have such "grid views":

<img width="622" alt="image" src="https://user-images.githubusercontent.com/2157566/162139401-9661af83-d301-486a-851b-5f5cdf879b57.png">

@CamachoDejay How do you currently achieve this?

I am asking, because we have some dedicated functionality to automatically put many images into a grid, but I am not sure we yet make this convenient in the ProjectCreator API, do we?
@K-Meech

**Comment by constantinpape** [comment]
@KateMoreva has already put some work into doing this programatically, but we haven't merged it yet, see #416.

**Comment by K-Meech** [comment]
Yes - there's nothing in project creator for grids yet. We'd need to merge @KateMoreva 's work first!

**Comment by CamachoDejay** [comment]
Hej @tischi,

At the moment I used the macro language (groovy) to accomplish this. Basically a user defined function that takes as input the x and y translation in microns. 
`
def load_section(idx, dataDir, datasetName, uiSelectionGroup, xtranslate, ytranslate){
	//idx is unique to my application naming convention
	sectionsPath = dataDir + "Slide_"+idx+".czi";
       // I use your nice helper to get the data as an image plus
	imp = MoBIEHelper.openWithBioFormats( sectionsPath, 0 );
	imp.show();

	channels = ChannelSplitter.split( imp );
	for ( c = 0; c < channels.length; c++ )
	{
		imageName = "Slide_"+idx+"_c" + c;
		affineTransform3D = new AffineTransform3D();
               // here comes the "grid"
		affineTransform3D.translate( new double[]{ xtranslate, ytranslate, 0} );
		images.addImage( channels[ c ], imageName, datasetName, ImageDataFormat.OmeZarr, 
                ProjectCreator.ImageType.image, affineTransform3D, uiSelectionGroup );
	}
	
	imp.close();
}
`
As you can see it is not fancy at all. Then I just had a helper function to create the x-y shifts in microns give the desired distance and the desired distribution e.g. 4x3

**Comment by tischi** [comment]
This is in fact, quite cool, we do essentially nothing else with the `grid` functionality.

**Comment by tischi** [comment]
Hi @KateMoreva @K-Meech,

FYI: There are now more users requesting the creation of grid views. I will thus try to look into this...if you there's anything I should know, please let me know.

**Comment by KateMoreva** [comment]
> Hi @KateMoreva @K-Meech,
> 
> FYI: There are now more users requesting the creation of grid views. I will thus try to look into this...if you there's anything I should know, please let me know.

Hi @tischi, 

I'll try to take a look at the code this weekend.

**Comment by tischi** [comment]
Hi @KateMoreva,

Nice! Thanks! You will have to rebase your code, because the `main` branch changed a lot 😄 !
And please develop against the `main` branch (not against `develop` as before).
